tangos/live_calculation/query_multivalue_folding.py

```diff
@@ -34,11 +35,25 @@ def refold(self, results_to_refold):
             mask.mark_nones_as_masked(determiner_slice)
             results_slice = mask.mask(results_slice.T).T
             determiner_slice = mask.mask(determiner_slice)
-            if len(determiner_slice)!=0:
+
+            truth_slice = np.full(len(determiner_slice),True).astype(np.bool)
+            for cc in self.constraints_columns:
+                constraint = results_slice[cc]
+                constraint = mask.mask(constraint)
+                constraint = constraint.astype(np.bool)
+                truth_slice[(constraint == False)|(constraint == None)] = False
+
+            if len(determiner_slice)!=0 and True in truth_slice:
+                mask2 = QueryMask()
+                mask2.mark_false_as_masked(truth_slice)
```

Comment:
This all seems quite complicated - could one actually just use the same mask object twice, i.e. allow the mask to be 'updated'?

Reply:
so really the only difference is no reinitialization of a new mask object and just do `mask.mark_fasle_as_masked(truth_slice)`, which would reset and create essentially a new mask because truth_slice is not the same length as the original masked array. Do I have that right?

Reply:
Yes, that's right... as currently implemented marking the mask loses the previous mask but that could be fixed so that they get &ed together

Reply:
hmmm yeah. I suppose that I would also have to make sure that the original `truth_slice` is also the same size as originally masked array. As it is now, truth_slice is initialized based on the first mask. But I can see how this is a cleaner way of doing it. I'll see if I can change the mask to be more versatile